### PR TITLE
test(ticdc): fix http api it and remove some duplicated script functions

### LINKED ISSUE #2958: http_api flaking test
### Which jobs are flaking?

kafka integration test: http_api

### Which test(s) are flaking?

[2021-09-28T15:18:29.092Z] Traceback (most recent call last):

[2021-09-28T15:18:29.092Z]   File "/home/jenkins/agent/workspace/cdc_ghpr_kafka_integration_test/go/src/github.com/pingcap/ticdc/tests/http_api/util/test_case.py", line 315, in <module>

[2021-09-28T15:18:29.092Z]     func(*sys.argv[2:])

[2021-09-28T15:18:29.092Z]   File "/home/jenkins/agent/workspace/cdc_ghpr_kafka_integration_test/go/src/github.com/pingcap/ticdc/tests/http_api/util/test_case.py", line 92, in pause_changefeed

[2021-09-28T15:18:29.092Z]     assert resp.status_code == rq.codes.accepted

[2021-09-28T15:18:29.092Z] AssertionError

### Jenkins logs or GitHub Actions link

```log
[2021-09-28T15:18:25.029Z] *** etcd info ***:

[2021-09-28T15:18:25.029Z] 

[2021-09-28T15:18:25.029Z] /tidb/cdc/capture/2d6267f7-a019-4d58-995b-262a9e89d335

[2021-09-28T15:18:25.029Z] 	{"id":"2d6267f7-a019-4d58-995b-262a9e89d335","address":"127.0.0.1:8300","version":"v5.3.0-alpha-57-gc9fbc70b"}

[2021-09-28T15:18:25.029Z] 

[2021-09-28T15:18:25.029Z] /tidb/cdc/owner/22317c2cfbecc570

[2021-09-28T15:18:25.029Z] 	2d6267f7-a019-4d58-995b-262a9e89d335'

[2021-09-28T15:18:25.029Z] + grep -q 'etcd info'

[2021-09-28T15:18:25.029Z] + break

[2021-09-28T15:18:25.029Z] + set +x

[2021-09-28T15:18:26.938Z] pass test: check health

[2021-09-28T15:18:26.938Z] pass test: get status

[2021-09-28T15:18:27.196Z] pass test: create changefeed

[2021-09-28T15:18:28.574Z] pass test: list changefeed

[2021-09-28T15:18:28.833Z] pass test: get changefeed

[2021-09-28T15:18:29.092Z] Traceback (most recent call last):

[2021-09-28T15:18:29.092Z]   File "/home/jenkins/agent/workspace/cdc_ghpr_kafka_integration_test/go/src/github.com/pingcap/ticdc/tests/http_api/util/test_case.py", line 315, in <module>

[2021-09-28T15:18:29.092Z]     func(*sys.argv[2:])

[2021-09-28T15:18:29.092Z]   File "/home/jenkins/agent/workspace/cdc_ghpr_kafka_integration_test/go/src/github.com/pingcap/ticdc/tests/http_api/util/test_case.py", line 92, in pause_changefeed

[2021-09-28T15:18:29.092Z]     assert resp.status_code == rq.codes.accepted

[2021-09-28T15:18:29.092Z] AssertionError

[2021-09-28T15:18:51.030Z] make: *** [integration_test_kafka] Error 1
```

### Anything else we need to know

- Does this test exist for other branches as well?

- Has there been a high frequency of failure lately?

### PR DESCRIPTION
**Pull request by asddongmen**
### What problem does this PR solve?

Issue Number: close #2958

### What is changed and how it works?
1. Add a function named `check_changefeed_state` in `/test/integration_tests/_utils` that can check a changefeed state and its error message.
2. Remove other duplicated script functions like `check_changefeed_mark_error` and `check_changefeed_mark_failed` etc.

### Check List 

Tests 

 - Integration test

Code changes
- None 

Side effects
- None

Related changes

 - None

### Release note 

```release-note
 None
```

### CODE REVIEW

File: tests/integration_tests/_utils/check_changefeed_state [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+state=$(echo $info | jq -r '.state')
+if [[ ! "$state" == "$expected_state" ]]; then
+	echo "changefeed state $state does not equal to $expected_state"
+	exit 1
+fi
+message=$(echo $info | jq -r '.error.message')
+if [[ ! "$message" =~ "$error_msg" ]]; then
+	echo "error message '$message' is not as expected '$error_msg'"
+	exit 1
+fi
```
Comment (hi-rustin) [inline tests/integration_tests/_utils/check_changefeed_state:33]:
```suggestion
fi

```

**Review by hi-rustin — APPROVED** [review]
Great!

**Review by overvenus — APPROVED** [review]

**Comment by asddongmen** [comment]
it case `kafka_sink_error_resume` fails, I am working on it.

File: tests/integration_tests/kafka_sink_error_resume/run.sh [hunk 2]
```diff
@@ -51,9 +38,9 @@ function run() {
 	run_sql "CREATE table kafka_sink_error_resume.t2(id int primary key auto_increment, val int);" ${UP_TIDB_HOST} ${UP_TIDB_PORT}
 	run_sql "INSERT INTO kafka_sink_error_resume.t1 VALUES ();"
 
-	ensure $MAX_RETRIES check_changefeed_state $pd_addr $changefeed_id "error"
+	ensure $MAX_RETRIES check_changefeed_state $pd_addr $changefeed_id "error" "null" ""
```
Comment (zhaoxinyu) [inline tests/integration_tests/kafka_sink_error_resume/run.sh:41]:
```suggestion
	ensure $MAX_RETRIES check_changefeed_state $pd_addr $changefeed_id "error" "kafka sink injected error" ""
```

Comment (asddongmen) [reply tests/integration_tests/kafka_sink_error_resume/run.sh]: thanks~

**Review by zhaoxinyu — APPROVED** [review]

**Pull request merged**